app/models/story_parser.rb

```diff
@@ -213,861 +210,848 @@ def parse_chapters_into_story(location, chapter_contents, options = {})
         work = Work.new(work_params)
       else
         new_chapter = get_chapter_from_work_params(work_params)
-        work.chapters << set_chapter_attributes(work, new_chapter, location, options)
+        work.chapters << set_chapter_attributes(work, new_chapter)
       end
     end
-    return set_work_attributes(work, location, options)
+    set_work_attributes(work, location, options)
   end
 
-  # tries to create an external author for a given url
-  def parse_author(location,external_author_name,external_author_email)
-    #If e_email option value is present (archivist importing from somewhere not supported for auto autho grab)
-    #will have value there, otherwise continue as usual. If filled, just pass values to create or find external author
-    #Stephanie 8-1-2013
 
-    #might want to add check for external author name also here, steph 12/10/2013
-    if external_author_email.present?
-      return parse_author_common(external_author_email,external_author_name)
+  # Everything below here is protected and should not be touched by outside
+  # code -- please use the above functions to parse external works.
 
-    else
+  protected
+
+  # tries to create an external author for a given url
+  def parse_author(location, external_author_name, external_author_email)
```

Comment:
This is one of the methods I changed to add an error message (it contains the error that appears when you import for others without an author or email, and from a site where email addresses can't be detected)